taichi/transforms/auto_diff.cpp

```diff
@@ -1008,6 +1008,7 @@ class MakeAdjoint : public ADTransform {
 
   void visit(GlobalStoreStmt *stmt) override {
     // erase and replace with global load adjoint
+    TI_ASSERT(!stmt->dest->is<ExternalPtrStmt>());
```

Comment:
I think it's a bit hacky here because it only handles the `ExternalPtrStmt`  in one specific statement.  Maybe we can raise a `not supported` error for `ExternalPtrStmt` when visiting it.